### PR TITLE
fix(ingest): unify emit interface

### PR DESCRIPTION
**Pull request by hsheth2**
As a byproduct, this fixes the Airflow plugin for kafka.

Slack ref: https://datahubspace.slack.com/archives/CUMUWQU66/p1669862530239169?thread_ts=1666852127.927679&cid=CUMUWQU66

## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/api/entities/corpgroup/corpgroup.py [hunk 1]
```diff
@@ -81,10 +81,4 @@ def emit(
...
         """
         for mcp in self.generate_mcp():
-            if type(emitter).__name__ == "DatahubKafkaEmitter":
-                assert callback is not None
-                kafka_emitter = cast("DatahubKafkaEmitter", emitter)
-                kafka_emitter.emit(mcp, callback)
-            else:
-                rest_emitter = cast("DatahubRestEmitter", emitter)
-                rest_emitter.emit(mcp)
+            emitter.emit(mcp, callback)
```
Comment (treff7es) [inline metadata-ingestion/src/datahub/api/entities/corpgroup/corpgroup.py:84]: Previously callback was mandatory for Kafka Emitter, or at least it failed with an assertion if it was not set. 
Is it fine if we lose this?

Comment (treff7es) [reply metadata-ingestion/src/datahub/api/entities/corpgroup/corpgroup.py:84]: what is callback used in the Kafka emitter?

Comment (hsheth2) [reply metadata-ingestion/src/datahub/api/entities/corpgroup/corpgroup.py:84]: we've changed it so the callback is not mandatory for kafka

by default, the kafka emitter will log error messages

**Review by treff7es — APPROVED** [review]
lgtm

**Pull request merged**